src/AlgebraicGeometry/Schemes/AffineSchemes/Morphisms/Attributes.jl

```diff
@@ -183,7 +183,7 @@ function preimage(
   )
   X = domain(phi)
   Y = codomain(phi)
-  check && (issubset(Z, Y) || (Z = intersect(Y, Z)))
+  check && (is_subscheme(Z, Y) || (Z = intersect(Y, Z)))
```

Comment:
This is not fault of this PR, but I think this code is too clever for its own good and should be changed; don't do an assignment in what looks like a logical expression, one could mistake the `=` for a `==`. Instead do
```suggestion
  if check && !is_subscheme(Z, Y)
    Z = intersect(Y, Z)
  end
```
However, more seriously, I also wonder why it does that -- normally I would expect a failed `check` to trigger an error, not to "correct" the problem. I can't think of any other place we do this, and indeed in this PR multiple other `check`s are modified that all do an error (but I do not claim to have checked all place, neither in this PR nor in all of Oscar)

Reply:
Actually, at least according to [Julia docs](https://docs.julialang.org/en/v1/manual/control-flow/), it is perfectly fine to use assignment together with short-circuit evaluation:
```
julia> true && (x = (1, 2, 3))
(1, 2, 3)
```

About `check`: we can take the preimage of any set by first taking the intersection with the codomain, so it looks OK to me. The user has the option to set `check` to `false` if they know that `Z` is a subscheme and they wish to avoid checking the containment.

Reply:
I don't think the question was whether this is valid julia code or not.

It is a question about how much time someone has to spend to understand this, when the code has to be improved/rewritten/fixed in a few years, and the original author is not available.

Reply:
As Tommy explained, this is not about what is valid syntax, this is about what is maintainable.

Moreover, `check` everywhere else means "raise an error if the input is invalid", and here it now means something different. That's bad. For example, before one could set `check=true` while e.g. debugging to verify one has no bug in the code, and if there is no error, one has evidence for that, while if there is a bug, one may see an error and can debug it.

But here `check` silently papers over mistakes. So presumably no error will be raised, and the behavior may even differ between `check=true` and `check=false`. I don't think that's acceptable.

But of course this is not an issue with this PR, it is an issue with the existing code, so it can be fixed in a separate PR. But fixed it should be.

Reply:
I completely agree. The name of the variable (`check`) should clearly indicate what it is checking (`check_z_contained_in_y`). And there should be consistency whether the function returns an error whenever the check fails. It might be worth elaborating on this in the [OSCAR styleguide](https://docs.oscar-system.org/stable/DeveloperDocumentation/styleguide/) (it only briefly mentions input sanity checks).